hazelcast/src/main/java/com/hazelcast/map/impl/DefaultRecordStore.java

```diff
@@ -438,13 +439,23 @@ public int clear() {
     }
 
     @Override
-    public void reset() {
+    public void reset(boolean clearIndexes) {
         clearRecordsMap(Collections.<Data, Record>emptyMap());
         resetSizeEstimator();
         resetAccessSequenceNumber();
+        if (clearIndexes) {
+            clearIndexes();
+        }
         mapDataStore.clear();
     }
 
+    private void clearIndexes() {
```

Comment:
maybe you can add this method to index service. seems much belong to there.

Btw are you removing all indexes of this map in this partition ? Shouldn't it be only this partitions indexes to be removed?